src/Semantics.js

```diff
@@ -196,6 +196,41 @@ Semantics.prototype.checkActionDicts = function() {
   }
 };
 
+Semantics.prototype.toRecipe = function() {
+  var sb = new common.StringBuffer();
+  sb.append('(function() {\n  var grammarFn = ');
+  sb.append(this.grammar.toRecipe());
+  sb.append('  var grammar = grammarFn.call(this);\n');
+  sb.append('  var semantics = grammar.semantics()');
+
+  ['Operation', 'Attribute'].forEach(function(type) {
+    var semantics = this[type.toLowerCase() + 's'];
+    Object.keys(semantics).forEach(function(semName) {
+      var fullName = semName;
+      if (semantics[semName].formals.length > 0) {
+        fullName += '(' + semantics[semName].formals.join(', ') + ')';
+      }
+      sb.append('\n    .add' + type + '(' + JSON.stringify(fullName) + ', {');
+
+      var actions = semantics[semName].actionDict;
+      var followUp = false;
+      Object.keys(actions).forEach(function(actionName) {
+        if (actionName === '_default') {
+          return; // FIXME: check whether _default has been overwritten
+        }
+        sb.append(followUp ? ',' : (followUp = true, ''));
```

Comment:
I find this to be really tricky. Assuming I understand it correctly, it might be more straightforward to do it like this: `var arr = Object.keys(actions).map(function(actionName) { ... });`, followed by `arr.join(',')`?